awscli/customizations/ec2instanceconnect/websocket.py

```diff
@@ -155,7 +155,7 @@ def connect(self, url, user_agent=None):
         proxy_options = None
         proxy_url = environment.get("HTTP_PROXY") or environment.get("HTTPS_PROXY")
         no_proxy = environment.get("NO_PROXY")
-        if proxy_url and parsed_url.hostname not in no_proxy:
+        if proxy_url and (no_proxy is None or parsed_url.hostname not in no_proxy):
```

Comment:
To simplify this down we should be able to just add a default to the environment `get()`:
```python
no_proxy = environment.get("NO_PROXY", "")
```
and from there we would not need to expand the if statement at all.